### PR TITLE
[IO] Small NFC improvements

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

File: core/clingutils/src/TClingUtils.cxx [hunk 1]
```diff
@@ -1744,7 +1744,7 @@ bool ROOT::TMetaUtils::ExtractAttrIntPropertyFromName(const clang::Decl& decl,
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// FIXME: a function of ~300 lines!
+/// FIXME: a function of 450+ lines!
```
Comment (Axel-Naumann) [inline core/clingutils/src/TClingUtils.cxx:1747]: LOL :-)

File: core/foundation/src/TClassEdit.cxx [hunk 2]
```diff
@@ -535,16 +535,13 @@ ROOT::ESTLType TClassEdit::STLKind(std::string_view type)
...
 
    // kind of stl container
-   auto len = type.length();
-   if (len) {
-      len -= offset;
-      for(int k=1;stls[k];k++) {
-         if (len == stllen[k]) {
-            if (type.compare(offset,len,stls[k])==0) return values[k];
-         }
+   const auto len = type.length() - offset;
```
Comment (Axel-Naumann) [inline core/foundation/src/TClassEdit.cxx:538]: How do we survive `TClassEdit::STLKind("")`?

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

File: core/foundation/src/TClassEdit.cxx (hunk 2 above)
Comment (eguiraud) [reply core/foundation/src/TClassEdit.cxx:538]: same as before? `len` will be 0, we enter the loop but nothing matches, we end up returning `ROOT::kNotSTL`.

I can add an early return if you prefer

File: core/clingutils/src/TClingUtils.cxx (hunk 1 above)
Comment (eguiraud) [reply core/clingutils/src/TClingUtils.cxx:1747]: nothing worse than a comment out of sync with the code :P

File: core/foundation/src/TClassEdit.cxx (hunk 2 above)
Comment (pcanal) [reply core/foundation/src/TClassEdit.cxx:538]: > we enter the loop but nothing matches

Then why go through the loop?  I am indeed in favor of adding early return both before and after the offset calculation.

Comment (eguiraud) [reply core/foundation/src/TClassEdit.cxx:538]: ok -- why after?

Comment (pcanal) [reply core/foundation/src/TClassEdit.cxx:538]: For "error case" like `"const "` and `"const std::"` .. probably mostly useless but cheap?

Comment (eguiraud) [reply core/foundation/src/TClassEdit.cxx:538]: alright!

Comment (Axel-Naumann) [reply core/foundation/src/TClassEdit.cxx:538]: And I was just confused (but appreciate the early returns = clarification in code!)

Comment (eguiraud) [reply core/foundation/src/TClassEdit.cxx]: added a commit with the improvement

Comment (eguiraud) [reply core/foundation/src/TClassEdit.cxx]: ...aaaand I screwed up, now fixing

File: core/foundation/src/TClassEdit.cxx [hunk 3]
```diff
@@ -510,9 +510,14 @@ bool TClassEdit::TSplitType::IsTemplate()
 
 ROOT::ESTLType TClassEdit::STLKind(std::string_view type)
 {
+   const auto len = type.length() - offset;
+   if (len == 0)
+      return ROOT::kNotSTL;
    size_t offset = 0;
    if (type.compare(0,6,"const ")==0) { offset += 6; }
    offset += StdLen(type.substr(offset));
+   if (len - offset == 0)
```
Comment (pcanal) [inline core/foundation/src/TClassEdit.cxx:519]:
```suggestion
   if ((len - offset) == 0)
```

File: core/foundation/src/TClassEdit.cxx (hunk 2 above)
Comment (eguiraud) [reply core/foundation/src/TClassEdit.cxx]: alright, done

**Review by pcanal — APPROVED** [review]

**Comment by pcanal** [comment]
Thanks.

**Pull request merged**